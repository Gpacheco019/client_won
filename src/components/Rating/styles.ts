import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  width: 100%;
`

export const LabelRate = styled.label`
  display: flex;
  margin-right: 3.2rem;
  cursor: pointer;

  :last-child {
    margin-right: 0;
  }
`

export const StarWrapper = styled.span.attrs({
  'data-testid': 'wrapper-star'
})`
  display: flex;
`

export const ValueRate = styled.input.attrs({
  'data-testid': 'value-rate'
})`
  display: none;
`
