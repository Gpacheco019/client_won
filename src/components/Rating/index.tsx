import { useState, ChangeEvent } from 'react'
import Star from '../Star'
import * as S from './styles'

export type RatingProps = {
  count?: number
  onChange(event: ChangeEvent<HTMLInputElement>): void
}

const Rating = ({ count = 5, onChange }: RatingProps) => {
  const [rating, setRating] = useState(0)

  const handleRating = (valueRating: number) => () => {
    setRating(valueRating)
  }

  return (
    <S.Wrapper>
      {Array(count)
        .fill(0)
        .map((_item, index) => {
          const valueRating = (index += 1)
          return (
            <S.LabelRate key={index}>
              <S.ValueRate
                value={valueRating}
                type="radio"
                name="rating"
                onClick={handleRating(valueRating)}
                onChange={onChange}
              />
              <S.StarWrapper>
                <Star
                  width={30}
                  height={28}
                  color={valueRating <= rating ? 'yellow' : 'gray'}
                />
              </S.StarWrapper>
            </S.LabelRate>
          )
        })}
    </S.Wrapper>
  )
}

export default Rating
