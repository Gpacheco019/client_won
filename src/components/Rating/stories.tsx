import { Story, Meta } from '@storybook/react'
import Rating, { RatingProps } from '.'

export default {
  title: 'Rating',
  component: Rating
} as Meta

export const Default: Story<RatingProps> = (args) => <Rating {...args} />
