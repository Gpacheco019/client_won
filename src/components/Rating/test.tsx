import { fireEvent, screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import Rating from '.'

const onChangeMock = jest.fn()

describe('<Rating />', () => {
  it('should render the rating correctly', () => {
    const { container } = renderWithTheme(
      <Rating count={5} onChange={onChangeMock} />
    )

    expect(screen.getAllByRole('img')).toHaveLength(5)
    expect(screen.getAllByTestId('value-rate')).toHaveLength(5)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render five stars default if not passed value', () => {
    renderWithTheme(<Rating onChange={onChangeMock} />)
    expect(screen.getAllByRole('img')).toHaveLength(5)
  })

  it('should called correctly click', () => {
    renderWithTheme(<Rating onChange={onChangeMock} />)

    const input = screen.getAllByTestId('value-rate')[0]
    const svg = screen.getAllByTestId('wrapper-star')[0].firstChild
    fireEvent.click(input)

    expect(onChangeMock).toHaveBeenCalledTimes(1)
    expect(svg).toHaveStyle({ color: '#FFD12D' })
  })
})
