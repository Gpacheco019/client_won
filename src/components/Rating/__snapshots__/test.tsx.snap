// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should render the rating correctly 1`] = ` `
